nixos/modules/services/x11/xidlehook.nix

```diff
@@ -0,0 +1,145 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xserver.xidlehook;
+in
+  {
+    options = {
+      services.xserver.xidlehook = {
+        enable = mkEnableOption "xidlehook";
+
+        notWhenFullscreen = mkOption {
+          default = true;
+          description = "Whether to lock the screen when an application is fullscreened.";
+          type = types.bool;
+        };
+
+        notWhenAudio = mkOption {
+          default = false;
+          description = "Whether to lock the screen when audio is playing.";
+          type = types.bool;
+        };
+
+        socket = mkOption {
+          default = ".local/xidlehook.socket";
+          description = ''
+            Socket file for controlling the daemon. The path is relative to the
+            user's home directory. Set to null to not have one.
+          '';
+          type = types.nullOr types.str;
+        };
+
+        timerPkgs = mkOption {
+          default = with pkgs; [xorg.xrandr gawk i3lock-fancy-rapid];
+          description = "Packages added to PATH for the timer commands.";
+        };
+
+        timers = mkOption {
+          default = let brightnessCmd = tgt:
+            "xrandr --output $(xrandr | awk '/ connected/{print $1}') --brightness ${toString tgt} ; " +
+            "echo brightness ${toString tgt}"; in
+          [
+            {
+              seconds = 5*60-5;
+              command = brightnessCmd 0.5;
+              canceller = brightnessCmd 1;
+            }
+            {
+              seconds = 5;
+              command = brightnessCmd 1 + " ; i3lock-fancy-rapid 10 5";
+            }
+          ];
+          description = ''
+            List of commands to run after the screen has been idle for a given \
+            amount of time. N.B. times are cumulative here - in the example \
+            below the first command runs after the screen is idle for 60 \
+            seconds and the second command runs after the screen has been idle \
+            for 70 seconds.
```

Comment:
```suggestion
            List of commands to run after the screen has been idle for a given
            amount of time. N.B. times are cumulative here - in the example
            below the first command runs after the screen is idle for 60
            seconds and the second command runs after the screen has been idle
            for 70 seconds.
```

Reply:
`N.B.`?

Reply:
Nota bene, it's a Latin abbreviation meaning "note well". I'll change it, since it's not clear to everyone.